### PR TITLE
Add section about stray files

### PR DESCRIPTION
**Pull request by sigurdm**
In support of https://github.com/dart-lang/pub/pull/4446

### CODE REVIEW

File: src/content/guides/workspaces.md [hunk 1]
```diff
@@ -109,6 +109,46 @@ To use pub workspaces, all your workspace packages (but not your dependencies)
 must have a SDK version constraint of `^3.6.0` or higher.
 :::
 
+<a name='stray-files'></a>
+## Stray files
+
+When you are migrating an existing mono-repo to use Pub workspaces, there will
```
Comment (MaryaBelanger) [inline src/content/guides/workspaces.md]:
```suggestion
When you are migrating an existing monorepo to use pub workspaces, there will
```

File: src/content/guides/workspaces.md [hunk 2]
```diff
@@ -109,6 +109,46 @@ To use pub workspaces, all your workspace packages (but not your dependencies)
...
+be existing "stray" `pubspec.lock` and `.dart_tool/package_config.json` files
+adjacent to each pubspec. These would shadow the `pubspec.lock` and
+`.dart_tool/package_config.json` placed next to the root
```
Comment (MaryaBelanger) [inline src/content/guides/workspaces.md]:
```suggestion
`.dart_tool/package_config.json` placed next to the root.
```

File: src/content/guides/workspaces.md [hunk 3]
```diff
@@ -109,6 +109,46 @@ To use pub workspaces, all your workspace packages (but not your dependencies)
...
+
+Therefore `pub get` will delete any `pubspec.lock` and
+`.dart_tool/package_config.json` located in directories between the root and
+(including) any workspace package.
```
Comment (MaryaBelanger) [inline src/content/guides/workspaces.md]:
```suggestion
`.dart_tool/package_config.json` located in directories between the root,
including any workspace package.
```

Does that keep the same meaning? I think the "and" might be redudant but you can make the call.

File: src/content/guides/workspaces.md [hunk 4]
```diff
@@ -109,6 +109,46 @@ To use pub workspaces, all your workspace packages (but not your dependencies)
...
 
+<a name='stray-files'></a>
+## Stray files
+
+When you are migrating an existing mono-repo to use Pub workspaces, there will
+be existing "stray" `pubspec.lock` and `.dart_tool/package_config.json` files
+adjacent to each pubspec. These would shadow the `pubspec.lock` and
+`.dart_tool/package_config.json` placed next to the root
+
+Therefore `pub get` will delete any `pubspec.lock` and
```
Comment (MaryaBelanger) [inline src/content/guides/workspaces.md]:
```suggestion
Therefore, `pub get` will delete any `pubspec.lock` and
```

File: src/content/guides/workspaces.md [hunk 5]
```diff
@@ -109,6 +109,46 @@ To use pub workspaces, all your workspace packages (but not your dependencies)
...
+    pubspec.lock                     # Deleted by `pub get`
+    .dart_tool/package_config.json   # Deleted by `pub get`
+    foo/
+      pubspec.yaml                   # Workspace member
+      pubspec.lock                   # Deleted by `pub get`
+      .dart_tool/package_config.json # Deleted by `pub get`
+```
+
+If any directory between the workspace root and a workspace package contains a
+"stray" `pubspec.yaml` file that is not member of the workspace, `pub get` will
```
Comment (MaryaBelanger) [inline src/content/guides/workspaces.md]:
```suggestion
"stray" `pubspec.yaml` file that is not a member of the workspace, `pub get` will
```

**Review by MaryaBelanger — APPROVED** [review]
This is really well written and easy to understand, thannks @sigurdm!

**Comment by parlough** [comment]
Thanks for this @sigurdm!

Would you mind rebasing this on `main` and opening a new upstream PR with this change?

**Comment by sigurdm** [comment]
Closed in favor of https://github.com/dart-lang/site-www/pull/6274

**Pull request closed without merging**